perf/tools/aggregate.lua

```diff
@@ -0,0 +1,93 @@
+-- File to aggregate the benchmark results from JSON files to the
+-- format parsable by the InfluxDB line protocol [1]:
+-- <measurement>,<tag_set> <field_set> <timestamp>
+--
+-- <tag_set> and <field_set> have the following format:
+-- <key1>=<value1>,<key2>=<value2>
+--
+-- The reported tag set is a set of values that can be used for
+-- filtering data (i.e., branch or benchmark name).
+--
+-- The script accepts the following parameters:
+--
+-- <input_dir> -- the directory from which the .json files are
+--                taken.
+-- <output>    -- the filename where the results are saved.
+--
+-- [1]: https://docs.influxdata.com/influxdb/v1/write_protocols/line_protocol_tutorial/
+
+local json = require('json')
+local fio = require('fio')
+
+local params = require('internal.argparse').parse(arg, {
+    {'input_dir', 'string'},
+    {'output', 'string'},
+})
+
+local input_dir = params.input_dir
+assert(input_dir and fio.path.is_dir(input_dir),
+       'given input_dir is not a directory')
+
+local output = params.output
+local out_fh = assert(io.open(output, 'w'))
```

Comment:
Since we anyway use `fio` in this script, I would use it for all the file operations. `assert(fio.open(output, {'O_WRONLY', 'O_CREAT', 'O_TRUNC'})` in this case.

`O_RDONLY` below in `read_all`.